### PR TITLE
Add CPMap sample [HZ-4175]

### PR DESCRIPTION
**Pull request by arodionov**
Add a CPMap sample into enterprise samples

### CODE REVIEW

**Comment by arodionov** [comment]
run-lab-run

File: enterprise/cp-map/src/main/java/com/hazelcast/codesamples/cp/cpmap/CPMember.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        String key = "key";
+        // Use FencedLock to set the initial value in an atomic way
+        lock.lock();
+        try {
+            if (cpMap.get(key) == null) {
+                cpMap.set(key, 0);
+            }
+        } finally {
+            lock.unlock();
+        }
```
Comment (gbarnett-hz) [inline enterprise/cp-map/src/main/java/com/hazelcast/codesamples/cp/cpmap/CPMember.java:38]:
```suggestion
        cpMap.putIfAbsent(key, 0);
```

File: enterprise/cp-map/src/main/java/com/hazelcast/codesamples/cp/cpmap/CPMember.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package com.hazelcast.codesamples.cp.cpmap;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.cp.CPMap;
+import com.hazelcast.cp.lock.FencedLock;
```
Comment (gbarnett-hz) [inline enterprise/cp-map/src/main/java/com/hazelcast/codesamples/cp/cpmap/CPMember.java:7]:
```suggestion
```

**Review by gbarnett-hz — COMMENTED** [review]
Suggestions for use with `putIfAbsent` which is now merged. If you're brave you could just commit the suggestions...

**Review by gbarnett-hz — APPROVED** [review]

**Pull request merged**